apiserver/logsink.go

```diff
@@ -16,23 +15,92 @@ import (
 	"github.com/juju/utils"
 	"github.com/juju/version"
 	"golang.org/x/net/websocket"
+	"gopkg.in/juju/names.v2"
 	"gopkg.in/natefinch/lumberjack.v2"
 
 	"github.com/juju/juju/apiserver/common"
 	"github.com/juju/juju/apiserver/params"
 	"github.com/juju/juju/state"
 )
 
-func newLogSinkHandler(h httpContext, w io.Writer) http.Handler {
-	return &logSinkHandler{ctxt: h, fileLogger: w}
+type LoggingStrategy interface {
+	Authenticate(*http.Request) error
+	Start()
+	Log(params.LogRecord) bool
+	Stop()
 }
 
-func newLogSinkWriter(logDir string) (io.WriteCloser, error) {
-	logPath := filepath.Join(logDir, "logsink.log")
+type AgentLoggingStrategy struct {
```

Comment:
No, good point. In fact all of this is internal to the package. I left the interface and its methods as public because it feels weird to define a private interface.